cmd/subctl/aws.go

```diff
@@ -0,0 +1,80 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subctl
+
+import (
+	"github.com/spf13/cobra"
+	cpaws "github.com/submariner-io/cloud-prepare/pkg/aws"
+	"github.com/submariner-io/submariner-operator/internal/cli"
+	"github.com/submariner-io/submariner-operator/internal/exit"
+	cloudaws "github.com/submariner-io/submariner-operator/pkg/cloud/aws"
+	"github.com/submariner-io/submariner-operator/pkg/cloud/cleanup"
+	"github.com/submariner-io/submariner-operator/pkg/cloud/prepare"
+)
+
+var (
+	awsConfig cloudaws.Config
+
+	awsPrepareCmd = &cobra.Command{
+		Use:   "aws",
+		Short: "Prepare an OpenShift AWS cloud",
+		Long:  "This command prepares an OpenShift installer-provisioned infrastructure (IPI) on AWS cloud for Submariner installation.",
+		Run: func(cmd *cobra.Command, args []string) {
+			var err error
+			if awsConfig.OcpMetadataFile != "" {
+				awsConfig.InfraID, awsConfig.Region, err = cloudaws.ReadFromFile(awsConfig.OcpMetadataFile)
+				exit.OnErrorWithMessage(err, "Failed to read AWS information from OCP metadata file")
+			} else {
+				expectFlag(infraIDFlag, awsConfig.InfraID)
+				expectFlag(regionFlag, awsConfig.Region)
+			}
+
+			err = prepare.AWS(&restConfigProducer, cloudPorts, &awsConfig, cli.NewReporter())
+			exit.OnError(err)
+		},
+	}
+
+	awsCleanupCmd = &cobra.Command{
+		Use:   "aws",
+		Short: "Clean up an AWS cloud",
+		Long: "This command cleans up an OpenShift installer-provisioned infrastructure (IPI) on AWS-based" +
+			" cloud after Submariner uninstallation.",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := cleanup.AWS(&restConfigProducer, cli.NewReporter())
```

Comment:
`cleanup` also needs `infraid` and `region` flags just like `prepare` and those value would need to be passed on.

Reply:
Done